src/HTTP.jl

```diff
@@ -37,24 +37,44 @@ include("sniff.jl")                    ;using .Sniff
 include("multipart.jl")                ;using .Forms
 include("Parsers.jl")                  ;import .Parsers: Headers, Header,
                                                          ParseError
-include("ConnectionPool.jl")           ;using .ConnectionPool
+include("Connections.jl")              ;using .Connections
+# backwards compat
+const ConnectionPool = Connections
 include("StatusCodes.jl")              ;using .StatusCodes
 include("Messages.jl")                 ;using .Messages
 include("cookies.jl")                  ;using .Cookies
 include("Streams.jl")                  ;using .Streams
+
+getrequest(r::Request) = r
+getrequest(s::Stream) = s.message.request
+
+# Wraps client-side "layer" to track the amount of time spent in it as a request is processed.
+function observe_layer(f)
+    function observelayer(req_or_stream; kw...)
```

Comment:
nitpick: `observe_layer` and `observelayer` is gonna drive me crazy

Also, the convention so far in the package is that these layer function are named in squashedgermaniccase (unfortunately imo) like `headerslayer, timeoutlayer, dragonslayer, ...`, so we should probably stick to that for a more consistent user interface.

And then (in this PR) we're also giving the inner function a (different) name, like `defaultheaders` (or another noun name, like `timeouts`)

So think these should be named something like:

```suggestion
function observelayer(f)
    function observation(req_or_stream; kw...)
```